build: fix the build failures of Windows HIP release job

lcy0321 opened the pull request:
- Update the cache keys to include the HIP SDK version, preventing the use of outdated ROCm installation caches.
- Sync changes from release.yml to build.yml:
  - Update HIP SDK version to 25.Q3 and ROCm version to 6.4.2
  - Update the cache keys to reflect the new versions
-  Remove Windows HIP release for gfx1151 since the current stable rocWMMA does not support gfx1151.

lcy0321 committed:
build: fix the cache keys for Windows HIP release job

Update the cache keys to include the HIP SDK version, preventing the
use of outdated ROCm installation caches.

slaren commented on .github/workflows/release.yml:
```diff
@@ -543,21 +548,19 @@ jobs:
-          key: rocm-6.1-${{ runner.os }}-v1
-          restore-keys: |
-            rocm-6.1-${{ runner.os }}-
+          key: rocm-${{ env.HIPSDK_INSTALLER_VERSION }}-${{ runner.os }}
```
Can you update this as well in `build.yml`?

lcy0321 committed:
build: sync changes from release.yml to build.yml

- Update HIP SDK version to 25.Q3 and ROCm version to 6.4.2
- Update the cache keys to reflect the new versions

lcy0321 committed:
build: remove Windows HIP release for gfx1151
since the current stable rocWMMA does not support gfx1151.

slaren commented:
>build: remove Windows HIP release for gfx1151
>since the current stable rocWMMA does not support gfx1151.

Can we use the unstable version then? I would rather accept some instability than remove Strix Halo support. Support was already merged in May in https://github.com/ROCm/rocWMMA/pull/538
